tf_quant_finance/black_scholes/vanilla_prices.py

```diff
@@ -90,9 +105,17 @@ def option_price(volatilities,
   Raises:
     ValueError: If both `forwards` and `spots` are supplied or if neither is
       supplied.
+    ValueError: If both `risk_free_rates` and `discount_factors` is supplied.
+    ValueError: If `cost_of_carries` is supplied without `discount_factors`.
   """
   if (spots is None) == (forwards is None):
     raise ValueError('Either spots or forwards must be supplied but not both.')
+  if risk_free_rates and discount_factors:
+    raise ValueError('At most one of risk_free_rates and discount_factors may '
+                     'be supplied')
+  if cost_of_carries and not risk_free_rates:
+    raise ValueError('cost_of_carries may only be supplied alongside the '
+                     'risk_free_rates')
```

Comment:
if you have `discount_factors`, 
risk_free_rates = -tf.math.log(discount_factors) / expiries